### ISSUE TITLE
🐛[BUG][/Combat] Day/Weather Magic Accuracy Bonuses Not Implemented

### ISSUE THREAD

**Issue by RenaviOnEden** [HorizonFFXI/HorizonXI-Issues#1774]
Previous bug report was closed by Frank, who later said he had misread the ticket and this had not been implemented yet.  This was on July 29th, over 2 months ago.  I have no way of reopening the previous report since I was not the one who closed it.  Since the issue hasn't been corrected, I figured I'd resubmit.  

What does this bug affect?
Day/Weather bonuses for Magic Accuracy

Steps to Reproduce
Could test by casting on a monster with known values and expected resist rates without and then with Obi on correct day/weather. You should see no magic accuracy change on current Horizon.  On retail, there would be a magic accuracy gain.  

Expected behavior
Day is +5 MACC
Weather is +5 MACC
Weather x2 would be +10 MACC
All 3 combined would be +15 MACC

Screenshots
None 

Setup Information
30221103_1

Additional context
https://luteff11.livedoor.blog/archives/42385969.html - Day/Weather/Obi MACC Testing from JP source.

More info from Jimmayus when it was originally found:

**Jimmayus — 12/07/2022 10:47 AM
https://luteff11.livedoor.blog/archives/42385969.html
FF11赤ソロ小冊子
戯言シリーズ5　～天候とか虚誘とか発生率とか～ : FF11赤ソロ小冊子
こんにちは(･ω･)ノ戯言4では魔命特化の学とスキル特化の黒でウォーの門のミミズLV124にウォータ連打しました。結果は魔命特化の学に軍配が上がりました。今回はその時の結果をベースに、学者ならではの補足を行いました。■補足部分・天候有り（豪雨の陣&水輪の帯)でのウォ
戯言シリーズ5　～天候とか虚誘とか発生率とか～ : FF11赤ソロ小冊子
weather and day effects have a magic accuracy associated bonus
basically +5 for each, so double weather + day is +15 total
Casey — 12/07/2022 10:56 AM
is it always active?
Jimmayus — 12/07/2022 11:05 AM
apparently same proc rate as damage
Casey — 12/07/2022 11:08 AM
proc independent of the damage or at the same time?
Jimmayus — 12/07/2022 8:14 PM
independent apparently

**Comment by nixyyyy** [comment]
https://github.com/LandSandBoat/server/pull/4060

I think this may fix the problem.

**Comment by nixyyyy** [comment]
It looks to me like we do have macc being added/subtracted for day/weather effects...

```-- Magic Accuracy from Day's Element.
local function magicAccuracyFromDayElement(actor, spellElement)
    local magicAcc = 0

    if
        spellElement ~= xi.magic.ele.NONE and
        (math.random(1, 100) <= 33 or actor:getMod(xi.combat.element.elementalObi[spellElement]) >= 1)
    then
        local dayElement = VanadielDayElement()

        -- Strong day.
        if dayElement == spellElement then
            magicAcc = magicAcc + 5

        -- Weak day.
        elseif dayElement == xi.combat.element.weakDay[spellElement] then
            magicAcc = magicAcc - 5
        end
    end

    return magicAcc
end

-- Magic Accuracy from Weather's Element.
local function magicAccuracyFromWeatherElement(actor, spellElement)
    local magicAcc = 0

    -- Calculate if weather bonus triggers.
    if
        spellElement ~= xi.magic.ele.NONE and
        (math.random(1, 100) <= 33 or actor:getMod(xi.combat.element.elementalObi[spellElement]) >= 1)
    then
        local actorWeather = actor:getWeather()

        -- Strong weathers.
        if actorWeather == xi.combat.element.strongSingleWeather[spellElement] then
            magicAcc = magicAcc + actor:getMod(xi.mod.IRIDESCENCE) * 5 + 5
        elseif actorWeather == xi.combat.element.strongDoubleWeather[spellElement] then
            magicAcc = magicAcc + actor:getMod(xi.mod.IRIDESCENCE) * 5 + 10

        -- Weak weathers.
        elseif actorWeather == xi.combat.element.weakSingleWeather[spellElement] then
            magicAcc = magicAcc - actor:getMod(xi.mod.IRIDESCENCE) * 5 - 5
        elseif actorWeather == xi.combat.element.weakDoubleWeather[spellElement] then
            magicAcc = magicAcc - actor:getMod(xi.mod.IRIDESCENCE) * 5 - 10
        end
    end

    return magicAcc
end
```
